### ISSUE TITLE
Using CORDEX data, where the CORDEX grid is larger than the expected grid for the domain

### ISSUE THREAD

**Issue by mo-acottrel** [ESMValGroup/ESMValCore#2025]
There appears to be an issue when using ESMValTool to analyse CORDEX data, where the CORDEX grid doesn’t exactly match the expected grid for the domain (i.e. per https://github.com/euro-cordex/tables/blob/main/domains/cordex.csv ). 

For example, considering a CORDEX model for the CAM-44 domain, which has a bigger latitude range (139 grid points) than the specified CORDEX domain grid (113 grid points), but otherwise looks to be on an identical grid, ESMValTool generates the following error:

File "/opt/scitools/conda/environments/esmvaltool-2.8.0/lib/python3.10/site-packages/esmvalcore/cmor/_fixes/cordex/cordex_fixes.py", line 193, in fix_metadata
    self._fix_rotated_coords(cube, domain, domain_info)
  File "/opt/scitools/conda/environments/esmvaltool-2.8.0/lib/python3.10/site-packages/esmvalcore/cmor/_fixes/cordex/cordex_fixes.py", line 144, in _fix_rotated_coords
    self._check_grid_differences(old_coord, new_coord)
  File "/opt/scitools/conda/environments/esmvaltool-2.8.0/lib/python3.10/site-packages/esmvalcore/cmor/_fixes/cordex/cordex_fixes.py", line 114, in _check_grid_differences
    diff = np.max(np.abs(old_coord.points - new_coord.points))
ValueError: operands could not be broadcast together with shapes (139,) (113,) 

Given such CORDEX data could be a valid CORDEX output (I think the specified domains are the minimum that the CORDEX experiments are required to generate), should ESMValTool be able to accept this data (e.g. by extracting a subset of the region to match the expected domain, before applying the _check_grid_differences check)?
[main_log_debug.txt](https://github.com/ESMValGroup/ESMValCore/files/11466120/main_log_debug.txt)
[recipe_test_cordex_grid.txt](https://github.com/ESMValGroup/ESMValCore/files/11466150/recipe_test_cordex_grid.txt)

**Comment by nhsavage** [comment]
@sloosvel @pepcos @zklaus - any thoughts on this one? I think that at the very least, the code should check that the length of the old_coord.points and new_coord.points are the same before differencing, and if they aren't raise a RecipeError as done when the points have too large a difference. Ideally at this point the code would first check if the new_coord is a superset of old_coord.points

I have put a copy of the files that fail on Jasmin in a group workspace, or you can get these from ESGF - example files are CAM-44/SMHI/NOAA-GFDL-GFDL-ESM2M/historical/r1i1p1/RCA4/v1/day/pr/v20170508/pr_CAM-44_NOAA-GFDL-GFDL-ESM2M_historical_r1i1p1_SMHI-RCA4_v1_day_19960101-20001231.nc and CAM-44/SMHI/NOAA-GFDL-GFDL-ESM2M/historical/r1i1p1/RCA4/v1/day/pr/v20170508/pr_CAM-44_NOAA-GFDL-GFDL-ESM2M_historical_r1i1p1_SMHI-RCA4_v1_day_20010101-20051231.nc

for reference the [CORDEX protocol ](https://cordex.org/wp-content/uploads/2020/05/cordex_general_instructions.pdf) states:

_Please be aware that the intention with the CORDEX domain is that it should be a common analysis region. In this sense the CORDEX domain should be fully inside your model domain and not include any of the boundary relaxation zone. We appreciate that domains will not be identical and for various reasons groups will choose to modify their actual domains somewhat. We ask that the CORDEX domain be fully within your interior model domain and you try to deviate as little as possible from the requested domain, while still satisfying your own interests._

**Comment by ghossh** [comment]
I could reproduce this issue when I'm using `use_standard_grid: true` in the recipe.

**Comment by bouweandela** [comment]
Indeed, this issue should be solved by setting `use_standard_grid: false`, which is available since #3059 and, optionally, regridding to the standard CORDEX domain if needed using #3096.
